File: app/src/main/java/App/repositories/retrofit/datasource/VentasRepository.java
package App.repositories.retrofit.datasource;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import App.repositories.retrofit.apiclient.RetrofitService;
import App.repositories.retrofit.responses.ListaAuxResponse;
import App.repositories.retrofit.responses.VentasResponse;
import App.repositories.retrofit.service.VentasService;
import App.repositories.room.entities.ListaAux;
import App.repositories.room.entities.Ventas;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;


public class VentasRepository {

    private static VentasRepository ventasRepository;
    private VentasService ventasService;

    public VentasRepository() {
        ventasService = RetrofitService.createService(VentasService.class);
    }

    public static VentasRepository getInstance() {
        if (ventasRepository == null) {
            ventasRepository = new VentasRepository();
        }
        return ventasRepository;
    }


    public MutableLiveData<List<Ventas>> getListaVentasRepository(int clienteId) {
        final MutableLiveData<List<Ventas>> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        params.put("clienteId", String.valueOf(clienteId));

        Call<VentasResponse> call = ventasService.getVentasCService(params);

        Log.d("response:", "Bien response ventas++");


        call.enqueue(new Callback<VentasResponse>() {
            @Override
            public void onResponse(Call<VentasResponse> call, Response<VentasResponse> response) {

                Log.d("response:", "Bien response ventas");
                Log.d("response:", "" + response.body().getStatus().toString());

                if (response.body().getStatus().equals("success")) {
                    data.setValue(response.body().getVentas());
                } else {
                    data.setValue(null);
                }

            }

            @Override
            public void onFailure(Call<VentasResponse> call, Throwable t) {

                data.setValue(null);

                Log.d("response:", "Mal response ventas");
                Log.d("response:", "" + t.getMessage());


            }
        });

        return data;

    }

    public MutableLiveData<List<ListaAux>> getListaEmpleadosRepository() {
        final MutableLiveData<List<ListaAux>> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        Call<ListaAuxResponse> call = ventasService.getEmpleadosService(params);

        //Log.d("empleado:","Bien response empleado++");


        call.enqueue(new Callback<ListaAuxResponse>() {
            @Override
            public void onResponse(Call<ListaAuxResponse> call, Response<ListaAuxResponse> response) {

                if (response.body().getStatus().equals("success")) {

                    data.setValue(response.body().getListaAux());
                    Log.d("response:", "Bien response ventas");

                    Log.d("response:", "" + response.body().getStatus().toString());
                    Log.d("empleado:", "" + response.body().getListaAux().get(1).id);


                } else {
                    data.setValue(null);
                    Log.d("empleado:", "Mal response ventas");
                }

            }

            @Override
            public void onFailure(Call<ListaAuxResponse> call, Throwable t) {

                data.setValue(null);

                Log.d("empleado:", "Mal response ventas");
                Log.d("response:", "" + t.getMessage());


            }
        });

        return data;

    }

    public MutableLiveData<List<ListaAux>> getListaTicoRepository() {
        final MutableLiveData<List<ListaAux>> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        Call<ListaAuxResponse> call = ventasService.getTipoCService(params);

        //Log.d("empleado:","Bien response empleado++");


        call.enqueue(new Callback<ListaAuxResponse>() {
            @Override
            public void onResponse(Call<ListaAuxResponse> call, Response<ListaAuxResponse> response) {

                if (response.body().getStatus().equals("success")) {

                    data.setValue(response.body().getListaAux());
                    //Log.d("response:","Bien response ventas");

                    //Log.d("response:",""+response.body().getStatus().toString());
                    //Log.d("tico:",""+response.body().getListaAux().get(1).id);


                } else {
                    data.setValue(null);
                    Log.d("tico:", "Mal response ventas");
                }

            }

            @Override
            public void onFailure(Call<ListaAuxResponse> call, Throwable t) {

                data.setValue(null);

                Log.d("tico:", "Mal response ventas");
                Log.d("response:", "" + t.getMessage());


            }
        });

        return data;

    }

    public MutableLiveData<Ventas> setVentasRepository(int maes_esta_conf,
                                                       String descripcion,
                                                       String fecha_llegada,
                                                       String fecha_salida,
                                                       int maes_tico,
                                                       int maes_tite,
                                                       int empl_id,
                                                       int clie_id) {

        final MutableLiveData<Ventas> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        params.put("maes_esta_conf", String.valueOf(maes_esta_conf));
        params.put("descripcion", String.valueOf(descripcion));
        params.put("fecha_llegada", String.valueOf(fecha_llegada));
        params.put("fecha_salida", String.valueOf(fecha_salida));
        params.put("maes_tico", String.valueOf(maes_tico));
        params.put("maes_tite", String.valueOf(maes_tite));
        params.put("empl_id", String.valueOf(empl_id));
        params.put("clie_id", String.valueOf(clie_id));

        Log.d("response:", "Registro confeccion: " + descripcion);

        Call<Ventas> call = ventasService.setVentaService(params);


        call.enqueue(new Callback<Ventas>() {
            @Override
            public void onResponse(Call<Ventas> call, Response<Ventas> response) {

                Log.d("response:", "Bien response ventas");
                Log.d("response:", "" + response.body().getStatus().toString());
                Log.d("response:", "" + response.body().getMessage().toString());

                if (response.body().getStatus().equals("success")) {
                    data.setValue(response.body());
                } else {
                    data.setValue(null);
                }

            }

            @Override
            public void onFailure(Call<Ventas> call, Throwable t) {

                data.setValue(null);

                Log.d("response:", "Mal response ventas");
                Log.d("response:", "" + t.getMessage());


            }
        });

        return data;

    }

    public MutableLiveData<List<ListaAux>> getListaTiteRepository() {
        final MutableLiveData<List<ListaAux>> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        Call<ListaAuxResponse> call = ventasService.getTipoTService(params);

        //Log.d("empleado:","Bien response empleado++");


        call.enqueue(new Callback<ListaAuxResponse>() {
            @Override
            public void onResponse(Call<ListaAuxResponse> call, Response<ListaAuxResponse> response) {

                if (response.body().getStatus().equals("success")) {

                    data.setValue(response.body().getListaAux());
                    //Log.d("response:","Bien response ventas");

                    //Log.d("response:",""+response.body().getStatus().toString());
                    //Log.d("tico:",""+response.body().getListaAux().get(1).id);


                } else {
                    data.setValue(null);
                    Log.d("tite:", "Mal response tite");
                }

            }

            @Override
            public void onFailure(Call<ListaAuxResponse> call, Throwable t) {

                data.setValue(null);

                Log.d("tite:", "Mal response tite");
                Log.d("response:", "" + t.getMessage());


            }
        });

        return data;

    }

    public MutableLiveData<Ventas> setDeleteVentasRepository(int ventasId) {
        final MutableLiveData<Ventas> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        params.put("ventasId", String.valueOf(ventasId));
        Log.d("deleteM", "Id" + ventasId);

        Call<Ventas> call = ventasService.deleteVentasService(params);

        call.enqueue(new Callback<Ventas>() {
            @Override
            public void onResponse(Call<Ventas> call, Response<Ventas> response) {

                Log.d("response:", "Bien response ventas eliminar");
                Log.d("response:","Status: "+response.body().getStatus().toString());

                data.setValue(response.body());



            }

            @Override
            public void onFailure(Call<Ventas> call, Throwable t) {

                data.setValue(null);

                Log.d("response:", "Mal response ventas eliminar");
                Log.d("response:", "Error: " + t.getMessage());
                Log.d("response:", "Error: " + t.getLocalizedMessage());


            }
        });

        return data;

    }


    public MutableLiveData<Ventas> setUpdateVentasRepository(int id, int maes_esta_conf, String descripcion, String fecha_llegada, String fecha_salida, int maes_tico, int maes_tite, int empl_id, int clie_id) {

        final MutableLiveData<Ventas> data = new MutableLiveData<>();

        final Map<String, String> params = new HashMap<>();

        params.put("ventasId", String.valueOf(id));
        params.put("maes_esta_conf", String.valueOf(maes_esta_conf));
        params.put("descripcion", String.valueOf(descripcion));
        params.put("fecha_llegada", String.valueOf(fecha_llegada));
        params.put("fecha_salida", String.valueOf(fecha_salida));
        params.put("maes_tico", String.valueOf(maes_tico));
        params.put("maes_tite", String.valueOf(maes_tite));
        params.put("empl_id", String.valueOf(empl_id));
        params.put("clie_id", String.valueOf(clie_id));

        Log.d("Update", "Id" + descripcion);

        Call<Ventas> call = ventasService.updateVentasService(params);

        call.enqueue(new Callback<Ventas>() {
            @Override
            public void onResponse(Call<Ventas> call, Response<Ventas> response) {

                Log.d("response:","Status: "+response.body().getStatus().toString());

                if (response.body().getStatus().equals("success")) {
                    data.setValue(response.body());
                } else {
                    data.setValue(null);
                }

            }

            @Override
            public void onFailure(Call<Ventas> call, Throwable t) {

                data.setValue(null);
                Log.d("response:", "Error: " + t.getMessage());

            }
        });

        return data;

    }
}

File: app/src/main/java/App/repositories/retrofit/datasource/VentasRepository.kt
package App.repositories.retrofit.datasource

import App.repositories.retrofit.apiclient.RetrofitService
import App.repositories.retrofit.responses.ListaAuxResponse
import App.repositories.retrofit.responses.VentasResponse
import App.repositories.retrofit.service.VentasService
import App.repositories.room.entities.ListaAux
import App.repositories.room.entities.Ventas
import android.util.Log
import androidx.lifecycle.MutableLiveData
import kotlinx.coroutines.flow.callbackFlow
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class VentasRepository {
    private val ventasService: VentasService?

    init {
        ventasService = RetrofitService.createService(VentasService::class.java)
    }

    fun getListaVentasRepository(clienteId: Int): MutableLiveData<List<Ventas?>?> {
        val data = MutableLiveData<List<Ventas?>?>()
        val params: MutableMap<String, String> = HashMap()
        params["clienteId"] = clienteId.toString()
        val call = ventasService!!.getVentasCService(params)
        Log.d("response:", "Bien response ventas++")

        call!!.enqueue(object : Callback<VentasResponse?> {
            override fun onResponse(
                call: Call<VentasResponse?>,
                response: Response<VentasResponse?>
            ) {
                Log.d("response:", "Bien response ventas")
                Log.d("response:", "" + response.body()!!.status.toString())
                if (response.body()!!.status == "success") {
                    data.setValue(response.body()!!.ventas)
                } else {
                    data.setValue(null)
                }
            }

            override fun onFailure(call: Call<VentasResponse?>, t: Throwable) {
                data.setValue(null)
                Log.d("response:", "Mal response ventas")
                Log.d("response:", "" + t.message)
            }
        })
        return data
    }


    //Log.d("empleado:","Bien response empleado++");
    val listaEmpleadosRepository: MutableLiveData<List<ListaAux?>?>
        get() {
            val data = MutableLiveData<List<ListaAux?>?>()
            val params: Map<String, String> = HashMap()
            val call = ventasService!!.getEmpleadosService(params)

            //Log.d("empleado:","Bien response empleado++");
            call!!.enqueue(object : Callback<ListaAuxResponse?> {
                override fun onResponse(call: Call<ListaAuxResponse?>, response: Response<ListaAuxResponse?>) {

                    if (response.body()?.status == "success") {
                        data.setValue(response.body()!!.listaAux)
                        Log.d("response:", "Bien response ventas")
                        Log.d("response:", "" + response.body()!!.status.toString())
                        Log.d("empleado:", "" + response.body()!!.listaAux!!.get(1)!!.id)

                    } else {

                        data.setValue(null)
                        Log.d("empleado:", "Mal response ventas")

                    }
                }

                override fun onFailure(call: Call<ListaAuxResponse?>, t: Throwable) {
                    data.setValue(null)
                    Log.d("empleado:", "Mal response ventas")
                    Log.d("response:", "" + t.message)
                }
            })
            return data
        }


    val listaTicoRepository: MutableLiveData<List<ListaAux?>?>
        get() {
            val data = MutableLiveData<List<ListaAux?>?>()
            val params: Map<String, String> = HashMap()
            val call = ventasService!!.getTipoCService(params)

            //Log.d("empleado:","Bien response empleado++");
            call!!.enqueue(object : Callback<ListaAuxResponse?> {
                override fun onResponse(call: Call<ListaAuxResponse?>, response: Response<ListaAuxResponse?>) {
                    if (response.body()?.status == "success") {
                        data.setValue(response.body()?.listaAux)
                        //Log.d("response:","Bien response ventas");

                        //Log.d("response:",""+response.body().getStatus().toString());
                        //Log.d("tico:",""+response.body().getListaAux().get(1).id);
                    } else {
                        data.setValue(null)
                        Log.d("tico:", "Mal response ventas")
                    }
                }

                override fun onFailure(call: Call<ListaAuxResponse?>, t: Throwable) {
                    data.setValue(null)
                    Log.d("tico:", "Mal response ventas")
                    Log.d("response:", "" + t.message)
                }
            })
            return data
        }

    fun setVentasRepository(maes_esta_conf: Int,
                            descripcion: String,
                            fecha_llegada: String,
                            fecha_salida: String,
                            maes_tico: Int,
                            maes_tite: Int,
                            empl_id: Int,
                            clie_id: Int): MutableLiveData<Ventas?> {
        val data = MutableLiveData<Ventas?>()
        val params: MutableMap<String, String> = HashMap()
        params["maes_esta_conf"] = maes_esta_conf.toString()
        params["descripcion"] = descripcion
        params["fecha_llegada"] = fecha_llegada
        params["fecha_salida"] = fecha_salida
        params["maes_tico"] = maes_tico.toString()
        params["maes_tite"] = maes_tite.toString()
        params["empl_id"] = empl_id.toString()
        params["clie_id"] = clie_id.toString()
        Log.d("response:", "Registro confeccion: $descripcion")
        val call = ventasService!!.setVentaService(params)
        call!!.enqueue(object : Callback<Ventas?> {
            override fun onResponse(call: Call<Ventas?>, response: Response<Ventas?>) {
                Log.d("response:", "Bien response ventas")
                Log.d("response:", "" + response.body()?.status.toString())
                Log.d("response:", "" + response.body()?.message.toString())
                if (response.body()?.status == "success") {
                    data.setValue(response.body())
                } else {
                    data.setValue(null)
                }
            }

            override fun onFailure(call: Call<Ventas?>, t: Throwable) {
                data.setValue(null)
                Log.d("response:", "Mal response ventas")
                Log.d("response:", "" + t.message)
            }
        })
        return data
    }


    val listaTiteRepository: MutableLiveData<List<ListaAux?>?>
        get() {
            val data = MutableLiveData<List<ListaAux?>?>()
            val params: Map<String, String> = HashMap()
            val call = ventasService!!.getTipoTService(params)

            //Log.d("empleado:","Bien response empleado++");
            call!!.enqueue(object : Callback<ListaAuxResponse?> {
                override fun onResponse(call: Call<ListaAuxResponse?>, response: Response<ListaAuxResponse?>) {
                    if (response.body()?.status == "success") {
                        data.setValue(response.body()?.listaAux)
                        //Log.d("response:","Bien response ventas");

                        //Log.d("response:",""+response.body().getStatus().toString());
                        //Log.d("tico:",""+response.body().getListaAux().get(1).id);
                    } else {
                        data.setValue(null)
                        Log.d("tite:", "Mal response tite")
                    }
                }

                override fun onFailure(call: Call<ListaAuxResponse?>, t: Throwable) {
                    data.setValue(null)
                    Log.d("tite:", "Mal response tite")
                    Log.d("response:", "" + t.message)
                }
            })
            return data
        }

    fun setDeleteVentasRepository(ventasId: Int): MutableLiveData<Ventas?> {
        val data = MutableLiveData<Ventas?>()
        val params: MutableMap<String, String> = HashMap()
        params["ventasId"] = ventasId.toString()
        Log.d("deleteM", "Id$ventasId")
        val call = ventasService!!.deleteVentasService(params)
        call!!.enqueue(object : Callback<Ventas?> {
            override fun onResponse(call: Call<Ventas?>, response: Response<Ventas?>) {
                Log.d("response:", "Bien response ventas eliminar")
                Log.d("response:", "Status: " + response.body()?.status.toString())
                data.setValue(response.body())
            }

            override fun onFailure(call: Call<Ventas?>, t: Throwable) {
                data.setValue(null)
                Log.d("response:", "Mal response ventas eliminar")
                Log.d("response:", "Error: " + t.message)
                Log.d("response:", "Error: " + t.localizedMessage)
            }
        })
        return data
    }

    fun setUpdateVentasRepository(id: Int, maes_esta_conf: Int, descripcion: String, fecha_llegada: String?, fecha_salida: String?, maes_tico: Int, maes_tite: Int, empl_id: Int, clie_id: Int): MutableLiveData<Ventas?> {
        val data = MutableLiveData<Ventas?>()
        val params: MutableMap<String, String> = HashMap()
        params["ventasId"] = id.toString()
        params["maes_esta_conf"] = maes_esta_conf.toString()
        params["descripcion"] = descripcion
        params["fecha_llegada"] = fecha_llegada.toString()
        params["fecha_salida"] = fecha_salida.toString()
        params["maes_tico"] = maes_tico.toString()
        params["maes_tite"] = maes_tite.toString()
        params["empl_id"] = empl_id.toString()
        params["clie_id"] = clie_id.toString()
        Log.d("Update", "Id$descripcion")

        val call = ventasService!!.updateVentasService(params)

        call!!.enqueue(object : Callback<Ventas?> {
            override fun onResponse(call: Call<Ventas?>, response: Response<Ventas?>) {
                Log.d("response:", "Status: " + response.body()?.status.toString())
                if (response.body()?.status == "success") {
                    data.setValue(response.body())
                } else {
                    data.setValue(null)
                }
            }

            override fun onFailure(call: Call<Ventas?>, t: Throwable) {
                data.setValue(null)
                Log.d("response:", "Error: " + t.message)
            }
        })
        return data
    }

    companion object {
        private var ventasRepository: VentasRepository? = null
        val instance: VentasRepository?
            get() {
                if (ventasRepository == null) {
                    ventasRepository = VentasRepository()
                }
                return ventasRepository
            }
    }
}